web_app/db/crud.py

```diff
@@ -618,25 +618,96 @@ def set_notification_allowed(
                 user_data = {
                     "telegram_id": telegram_id,
                     "wallet_id": wallet_id,
-                    "is_allowed_notification": True
+                    "is_allowed_notification": True,
                 }
                 return self.create_telegram_user(user_data)
-            
+
     def allow_notification(self, telegram_id: int) -> bool:
         """
         Update is_allowed_notification field to True for a specific telegram user
-        
+
         Args:
             telegram_id: Telegram user ID
-            
+
         Raises:
             ValueError: If the user with the given telegram_id is not found
         """
         with self.Session() as session:
-            user = session.query(TelegramUser).filter_by(telegram_id=telegram_id).first()
+            user = (
+                session.query(TelegramUser).filter_by(telegram_id=telegram_id).first()
+            )
             if not user:
                 raise ValueError(f"User with telegram_id {telegram_id} not found")
-            
+
             user.is_allowed_notification = True
             session.commit()
             return True
+
+
+class DepositDBConnector(DBConnector):
+    """
+    Provides database connection and operations management for the Vault model.
+    """
+
+    def create_vault(self, user: User, symbol: str, amount: str) -> Vault:
+        """
+        Creates a new vault instance
+
+        :param user: A user model instance
+        :param symbol: Token symbol or address
+        :param amount: An amount in string
+
+        :return: Vault
+        """
+        vault = Vault(user_id=user.id, symbol=symbol, amount=amount)
+        self.write_to_db(vault)
+        return vault
+
+    def get_or_create_vault(self, wallet_id: str, symbol: str) -> Vault | None:
```

Comment:
Please, rework this method into `get_vault` only
This method is overcomplicated

Reply:
Resolved.